data/scenarios/Tutorials/loops.yaml

```diff
@@ -0,0 +1,77 @@
+version: 1
+name: Loops
+description: |
+  Learn how to implement loops.
+objectives:
+  - goal:
+      - |
+        The previous tutorial taught you how to use conditional
+        expressions; this one will guide you through using loops.
+      - |
+        To the east of your base lies a path of `rock`{=entity}s.  The
+        path extends in a straight line for an unknown distance, and at
+        the end of the path lies a lump of `gold`{=entity}.  You must
+        program a robot to go and fetch it!
+      - |
+        Some kind of `while`{=snippet} loop would obviously come in
+        handy here. However, the Swarm language does not have loops,
+        only recursion, which is enabled by the `strange loop`{=entity}
+        device.  You can use recursion to implement your own control
+        structures.  For example, try filling in the `...`{=snippet}
+        below to implement your own while loop function:
+      - |
+        `def while : Cmd Bool -> Cmd Unit -> Cmd Unit = \cond. \body. ... end`{=snippet}
+      - |
+        This function should take a condition (a `Cmd Bool`{=type}, for
+        example, `ishere "rock"`) and a command, and execute the
+        command repeatedly as long as the condition returns `True`.
+    condition: |
+      try {as base {has "gold"}} {pure false}
+solution: |
+  def while : Cmd Bool -> Cmd a -> Cmd Unit = \test. \body.
+    b <- test;
+    if b {body; while test body} {}
+  end
+
+  def follow = while (ishere "rock") move end
+
+  build {
+    turn east; move;
+    follow;
+    grab; turn back; move;
+    follow;
+    give base "gold"
+  }
+robots:
+  - name: base
+    heavy: true
+    loc: [0, 0]
+    dir: north
+    display:
+      char: Ω
+      attr: robot
+    devices:
+      - logger
+      - 3D printer
+      - dictionary
+    inventory:
+      - [8, compass]
+      - [8, solar panel]
+      - [8, logger]
+      - [8, treads]
+      - [8, grabber]
```

Comment:
Embarrassingly, I failed two times:
1. I forgot to move out of gold and could not salvage it with a robot
2. When I restarted, I found that I can not salvage with base

Could we add a `toolkit` please? 😅

Reply:
Done!